### DISCUSSION TITLE
FrameBuffer.ellipse() missing

### DISCUSSION THREAD

**Discussion by MartinMohr** [micropython/micropython#9655] [Libraries & Drivers] [answered]
Hi all,
I have a strange Error in my Program  
AttributeError: 'FrameBuffer' object has no attribute 'ellipse'
The FrameBuffer Class should have this Attribute
https://docs.micropython.org/en/latest/library/framebuf.html
I am lost. What is going wrong?

**Comment by scruss** [comment, marked as answer]
This is a docs convention.

The *current* (v1.19.1; that is, the version you're likely to be running) [framebuf](https://docs.micropython.org/en/v1.19.1/library/framebuf.html) intro paragraph for framebuf.FrameBuffer:

> The FrameBuffer class provides a pixel buffer which can be drawn upon with pixels, lines, rectangles, text and even other FrameBuffer’s. It is useful when generating output for displays.

The *latest* (so for v1.20, or selected nightly builds on some platforms) [framebuf](https://docs.micropython.org/en/latest/library/framebuf.html) corresponding intro paragraph:

> The FrameBuffer class provides a pixel buffer which can be drawn upon with pixels, lines, rectangles, *ellipses, polygons,* text and even other FrameBuffers. It is useful when generating output for displays.

(emphasis mine).

*latest* is effectively **future** documentation. This catches a lot of users out.

The `MicroPython v1.19.1 on 2022-10-14; Raspberry Pi Pico with RP2040` build has `ellipse()`:

```
>>> import framebuf
>>> help(framebuf.FrameBuffer)
object <class 'FrameBuffer'> is of type type
  fill -- <function>
  fill_rect -- <function>
  pixel -- <function>
  hline -- <function>
  vline -- <function>
  rect -- <function>
  line -- <function>
  ellipse -- <function>
  poly -- <function>
  blit -- <function>
  scroll -- <function>
  text -- <function>
```

**Reply by jimmo** [reply]
Yeah I think we need to do something to make this more obvious -- some sort of banner on the "latest" version of the docs.

**Reply by MartinMohr** [reply]
Thanks for the fast reply ... ok I see.

MicroPython v1.19.1 on 2022-06-18; ESP32 module with ESP32
Type "help()" for more information.
>>> import framebuf
>>> help(framebuf.FrameBuffer)
object <class 'FrameBuffer'> is of type type
  fill -- <function>
  fill_rect -- <function>
  pixel -- <function>
  hline -- <function>
  vline -- <function>
  rect -- <function>
  line -- <function>
  blit -- <function>
  scroll -- <function>
  text -- <function>

**Reply by jimmo** [reply]
docs PR here: https://github.com/micropython/micropython/pull/9668

@MartinMohr You're running the 1.19.1 release version. You need to install a recent nightly build to get `ellipse`.

**Reply by MartinMohr** [reply]
Thanks
